init-distro-rid.sh

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+
+# initNonPortableDistroRid
+#
+# Input:
+#   isCrossBuild? (nullable vararg:int)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# initNonPortableDistroRid will attempt to initialize a non portable rid. These
+# rids are specific to distros need to build the product/package and consume
+# them on the same platform.
+#
+# Below is the list of current non-portable platforms.
+#
+# Builds from the following *must* be non-portable:
+#
+#   |    OS     |           Expected RID            |
+#   -------------------------------------------------
+#   |   rhel6   |           rhel.6-x64              |
+#   |  alpine*  |        linux-musl-(arch)          |
+#   |  freeBSD  |        freebsd.(version)-x64      |
+#
+# It is important to note that the function does not return, but it will set
+# __DistroRid if there is a non-portable distro rid to be used.
+#
+initNonPortableDistroRid()
+{
+    # Make sure out parameter is cleared.
+    export __DistroRid=
+
+    local nonPortableBuildID=""
+    local isCrossBuild=0
+    local crossBuildLocation=""
+
+    # If there is not an optional argument passed then this will be treated
+    # as if there is no crossbuild.
+    if [ -z "$1" ]; then
+        isCrossBuild=0
+    else
+        isCrossBuild=$1
+    fi
+
+    if (( ${isCrossBuild} == 1 )); then
+        crossBuildLocation=${ROOTFS_DIR}
+    fi
+
+    if [ "$__BuildOS" = "Linux" ]; then
+        if [ -e "${crossBuildLocation}/etc/redhat-release" ]; then
+            local redhatRelease=$(<${crossBuildLocation}/etc/redhat-release)
+
+            if [[ "${redhatRelease}" == "CentOS release 6."* || "$redhatRelease" == "Red Hat Enterprise Linux Server release 6."* ]]; then
```

Comment:
It seems you are missing the case for RHEL / CentOS >= 7. And also, looking at how the /etc/redhat-release varies per version (`CentOS Linux release 7.2.1511 (Core)` on my local CentOS 7 VM vs `CentOS release 6.*`) and the fact that the version is just encoded in the middle of a string, I would rather prefer using the /etc/os-release as the default location to look at and using the /etc/redhat-release as a fallback for version 6 only. I know I've already discussed this with @echesakov when he was working on a similar change and that I've agreed on using the /etc/redhat-release first, but I have reconsidered that.

Reply:
@janvorli per talking with @echesakovMSFT we came to a belief that centos-7 should never be set to a non-portable build. It should use the generic linux-x64 __DistroRid.

Reply:
I think that we should be able to build for any distro as non-portable. This will be the default build mode for distro owners wanting to have native packages of .NET core. For those, there is no point in making the build portable, it just adds unnecessary overhead.

Reply:
This has been addressed.